### PR TITLE
Experimental support for grpc tracing

### LINKED ISSUE #226: Context is dropped when using gRPC

### PR DESCRIPTION
**Pull request by matthewloring**
Fixes #226

### CODE REVIEW

**Comment by matthewloring** [comment]
@ofrobots PTAL.

File: lib/hooks/userspace/hook-grpc.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ */
+
+'use strict';
+
+var cls = require('../../cls.js');
+var shimmer = require('shimmer');
+var semver = require('semver');
+var agent;
+
+var SUPPORTED_VERSIONS = '^0.13.x';
```
Comment (ofrobots) [inline lib/hooks/userspace/hook-grpc.js]: Want to double check that we intend claim forward compatibility? We don't use `^` with most other libraries we support.

Comment (matthewloring) [inline lib/hooks/userspace/hook-grpc.js]: It is going to get broken sooner but unfortunately I think `0.13.x` is safer.

File: test/standalone/test-grpc-context.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      });
+    });
+    var server = app.listen(common.serverPort, function() {
+      var grpcServer = new grpc.Server();
+      grpcServer.addProtoService(test_proto.Tester.service, {
+        test: function(call, cb) {
+          cb(null, {message: 'world'});
+        }
+      });
+      grpcServer.bind('0.0.0.0:50051', grpc.ServerCredentials.createInsecure());
```
Comment (ofrobots) [inline test/standalone/test-grpc-context.js]: Bind to localhost only please. You don't want to open that port publicly even for the duration of the test.

Comment (ofrobots) [inline test/standalone/test-grpc-context.js]: nit: avoid hard-coded port number.

**Comment by ofrobots** [comment]
LGTM w/ comment, nit, and if your have verified that the test loses context without the grpc patching.

File: test/standalone/test-grpc-context.js (hunk 2 above)
Comment (matthewloring) [inline test/standalone/test-grpc-context.js]: Bound to localhost. We need to specify a port for grpc but I've moved it to a constant above.

**Comment by matthewloring** [comment]
Verified without patch.

**Pull request merged**